impeller/renderer/context.h

```diff
@@ -46,6 +46,8 @@ class Context {
   virtual std::shared_ptr<CommandBuffer> CreateTransferCommandBuffer()
       const = 0;
 
+  virtual bool HasThreadingRestrictions() const;
```

Comment:
nit: Can this name be a bit more descriptive? Maybe something like `CanUploadTextureFromAnyThread`?

It's not obvious what threading restrictions there are from this method name.

Reply:
Agreed. But you can call SetContents on the texture from any thread though. It's just that if there isn't a context available on that thread, it will be queued on one that has (which may be the main context). And, its not just texture upload, it could be any OpenGL call. So, this rather un-descriptive accessor.

Reply:
Agreed. But you can call SetContents on the texture from any thread though. It's just that if there isn't a context available on that thread, it will be queued on one that has (which may be the main context). And, its not just texture upload, it could be any OpenGL call. So, this rather un-descriptive accessor.

Reply:
I added a note to remove this API in the followup to this https://github.com/flutter/flutter/issues/104784.

Reply:
Sgtm